Dockerfile.updater

```diff
@@ -33,6 +33,7 @@ COPY --chown=dependabot:dependabot common ${DEPENDABOT_HOME}/common
 WORKDIR $DEPENDABOT_HOME/dependabot-updater
 
 RUN bundle config set --local path 'vendor' && \
+bundle config set --local frozen 'true' && \
```

Comment:
So IIUC this will ensure the built docker image has the updated `Gemfile.lock`, but is there a chance that copy of the `Gemfile.lock` will start to drift from the committed copy? IE, what ensures that as this version increases that the increase also makes it back to the committed code?? Should we also add a `git status --porcelain` somewhere?

Reply:
I didn't understand the question, which version do you mean when you say "this version increases", the one in `version.rb` files of every gem?

What the `frozen` configuration ensures is that whenever Bundler runs, it will abort if the `Gemfile.lock` is out of date with the `Gemfile`. In other words, that running Bundler won't cause any changes in the `Gemfile.lock` file.

Reply:
Another way to explain this, we're already using `aws-sdk-core 3.164.0` and other dependencies "upgraded" here, as can be seen in the [updater build logs](https://github.com/dependabot/dependabot-core/actions/runs/3319986944/jobs/5485825132) of the last build in the main branch:

```
Installing rubocop-ast 1.23.0
Fetching aws-sdk-core 3.164.0 (was 3.137.0)
Installing aws-sdk-core 3.164.0 (was 3.137.0)
Fetching webmock 3.18.1
```

This PR just makes this explicit and prevents this kind of divergence from happening in the future, but it shouldn't cause the updater to use any more up to date gems than the ones it's using now.

Reply:
Perfect, this code does exactly what I want than... I didn't understand the `frozen` flag... I was thinking that this caused bundler to regenerate the lockfile during the build step (aka `pip freeze` which i'm more familiar with). If it just means it regenerates the lock file and compares and flags divergence and throws an error, then it is exactly what we want of preventing drift.